### PR TITLE
sof-tplgreader: don't use RE for pipeline filtering

### PR DESCRIPTION
**Pull request by marc-hb**
If we have two pipelines with id=1 and id=10, current
RE is not able to filter these two pipelines from each
other.

This patch deprecates RE and use string comparison for
pipeline filtering.

v2 by Marc: don't add support for non-lists, see #713 review for
details.

Signed-off-by: Chao Song <chao.song@linux.intel.com>
Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

**Comment by marc-hb** [comment]
https://sof-ci.01.org/softestpr/PR715/build752/devicetest is all green.

File: tools/sof-tplgreader.py
Comment (fredoh9) [reply tools/sof-tplgreader.py:176]: @marc-hb Can you address this before 2 PM PST? So that we can merge this before daily cut-off time.

Comment (marc-hb) [reply tools/sof-tplgreader.py:176]: Sorry I missed this (there are too many other, unrelated pylint warnings). I'll rebase doc fix #714 and add it there OK? @plbossart said he needs this one ASAP to unblock Linux PRs.

**Comment by marc-hb** [comment]
https://sof-ci.01.org/softestpr/PR715/build754/devicetest is all green again.

File: tools/sof-tplgreader.py
Comment (fredoh9) [reply tools/sof-tplgreader.py:176]: #714 still need re, can you rebase this?

**Review by fredoh9 — APPROVED** [review]
Looks good!

**Pull request merged**
